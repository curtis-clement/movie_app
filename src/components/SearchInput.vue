<script setup lang="ts">

const props = withDefaults(defineProps<{
  disabled?: boolean;
  placeholder: string;
  modelValue: string;
}>(), {
  disabled: false,
});

const emit = defineEmits(['update:modelValue']);

</script>

<template>
  <input 
    type="text" 
    :placeholder="props.placeholder" 
    :value="props.modelValue" 
    :disabled="props.disabled"
    @input="emit('update:modelValue', ($event.target as HTMLInputElement).value)" 
  />
</template>

<style scoped>
input {
  border: 1px solid #ccc;
  border-radius: 4px;
  padding: 0.5rem;
  width: 10rem;
  min-width: 150px;
  margin-left: 1rem;
  margin-right: 0.5rem;
}

input:disabled {
  background-color: #f0f0f0;
  cursor: not-allowed;
  opacity: 0.6;
}
</style>