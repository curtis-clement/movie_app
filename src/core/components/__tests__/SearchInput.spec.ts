import { beforeEach, describe, it, expect } from 'vitest';
import { mount, VueWrapper } from '@vue/test-utils';
import SearchInput from '@/core/components/SearchInput.vue';

const mountComponent = (props: {
  placeholder: string;
  modelValue: string;
}) => {
  return mount(SearchInput, {
    props,
  });
};

describe('SearchInput', () => {
  let wrapper: VueWrapper<InstanceType<typeof SearchInput>>;

  beforeEach(() => {
    wrapper = mountComponent({
      placeholder: 'Search shows',
      modelValue: '',
    });
  });

  it('renders the component', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('passes props correctly', () => {
    expect(wrapper.props()).toEqual({
      placeholder: 'Search shows',
      modelValue: '',
      disabled: false,
    });
  });

  it('displays the placeholder text', () => {
    expect(wrapper.find('input').attributes('placeholder')).toBe('Search shows');
  });

  it('emits update:model-value event when input value changes', () => {
    const input = wrapper.find('input');
    input.setValue('New value');
    expect(wrapper.emitted('update:model-value')).toEqual([['New value']]);
  });

  it('does not emit update:modelValue event when input is disabled', async () => {
    wrapper.setProps({ disabled: true });
    await wrapper.vm.$nextTick();
    const input = wrapper.find('input');
    input.setValue('New value');
    expect(wrapper.emitted('update:model-value')).toBeUndefined();
  });
});
