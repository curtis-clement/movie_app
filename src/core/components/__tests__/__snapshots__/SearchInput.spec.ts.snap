// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchInput > renders the component 1`] = `"<input data-v-eb3440b3="" type="text" placeholder="Search shows" value="">"`;
